tensorflow/src/TensorFlow/Tensor.hs

```diff
@@ -189,3 +190,11 @@ instance TensorKind Ref where
 
 instance TensorKind Build where
     toBuild = id
+
+
+-- | Types which can be converted to `Tensor`.
+class ToTensor t where
```

Comment:
Would it be helpful to make `ToTensor` a superclass of `Rendered`. to reduce the signature of consumer functions like `gradients`?  Up to you.